src/hotspot/share/opto/parseHelper.cpp

```diff
@@ -34,6 +34,9 @@
 #include "opto/runtime.hpp"
 #include "runtime/sharedRuntime.hpp"
 
+unsigned peaNumMaterializations = 0;
```

Comment:
In convention, those statistical counters are defined with '#ifndef PRODUCT'.
`Parse::print_statistics` is compiled with that macro test.